web/api/v1/api.go

```diff
@@ -1526,7 +1530,70 @@ func (api *API) cleanTombstones(r *http.Request) apiFuncResult {
 	return apiFuncResult{nil, nil, nil, nil}
 }
 
-func (api *API) respond(w http.ResponseWriter, data interface{}, warnings storage.Warnings) {
+func (api *API) tryArrow(w http.ResponseWriter, r *http.Request, data interface{}, warnings storage.Warnings) error {
+	result, ok := data.(*queryData)
+	if !ok {
+		return fmt.Errorf("invalid result data type: %T", data)
+	}
+
+	if result.ResultType != parser.ValueTypeMatrix {
```

Comment:
As @roidelapluie mentioned, this could be done in some helper function in promQL package (?)?